tests/nested/manual/core20-early-config/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Test that gadget config defaults are applied early on core20.
+
+systems: [ubuntu-20.04-64]
+
+prepare: |
+    ENABLE_TPM="true"
+    ENABLE_SECURE_BOOT="true"
+    BUILD_SNAPD_FROM_CURRENT="true"
+    
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    mkdir extra-snaps
+
+    # Get the snakeoil key and cert
+    KEY_NAME=$(get_snakeoil_key)
+    SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+    SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+    # modify and repack gadget snap (add defaults section and install hook)
+    snap download --basename=pc --channel="20/edge" pc
+    unsquashfs -d pc-gadget pc.snap
+
+    cat defaults.yaml >> pc-gadget/meta/gadget.yaml
+    mkdir -p pc-gadget/meta/hooks
+    cp install pc-gadget/meta/hooks/
+
+    sbattach --remove pc-gadget/shim.efi.signed
```

Comment:
I would create 2 functions: 1 for unsign and 1 for sign the gadget in the nested.sh, so we have that logic just in 1 place.

Reply:
Right, makes sense

Reply:
Done.